### ISSUE TITLE
Follow operation should prefer the focused way

### ISSUE THREAD

**Issue by 1ec5** [openstreetmap/iD#8750]
The “follow” operation added in #8671 stops when encountering a node that’s shared among multiple features. iD already keeps track of the “focused” way for the purpose of reliably navigating back and forth among its vertices using the keyboard. It would be natural to extend this capability to the follow operation, so that the user doesn’t have to think as much about other connecting ways after choosing a way to follow. Invoking the follow operation would focus the way, and subsequent follow operations would choose that way from among the candidate ways.

https://github.com/openstreetmap/iD/blob/76990349bd2442a80b4c9bfb2440ea5c7a00a2d1/modules/modes/select.js#L150 https://github.com/openstreetmap/iD/blob/76990349bd2442a80b4c9bfb2440ea5c7a00a2d1/modules/modes/select.js#L48-L50

**Comment by sudhanshu2** [comment]
@mbrzakovic can I take this issue?

**Comment by Pho3b** [comment]
Hi and nice to meet you @mbrzakovic @1ec5 , i would like to try and work on this issue if possible.

**Comment by thyrhone** [comment]
is it still possible to work on this?

**Comment by 1ec5** [comment]
Thanks for your interest! There’s a fix available in #8978; perhaps you could help review it?

**Comment by sherryboxall** [comment]
Hello  @mbrzakovic @1ec5 ,

I would like to help work on this issue if possible. Thank you!

**Comment by Dustpie** [comment]
Can I still help on this issue if possible? Thank you!

**Comment by Gua00va** [comment]
Hey!! I want to work on this issue. I'm new here so I'm really excited to start contributing. Please assign it to me.

**Comment by k-yle** [comment]
This issue has already been fixed in two conflicting PRs: #8978 and #9340


I'm guessing the suprising interest in this issue is because it has a _"good first issue"_ label. @tyrasd it might be good to add this label to some other issues like #9452, #9217, #9448, #9441, #9440, #9312, #9402, #9400, ...

**Comment by 1ec5** [comment]
Right, GitHub has a feature that suggests issues labeled as https://github.com/openstreetmap/iD/labels/good%20first%20issue to users, but it doesn’t consider whether there are outstanding PRs already linked to them.

**Comment by DujaOSM** [comment]
Objection, your honor!
Implementing this will break workflow in a very common use case where we fill up the space with landuse polygons. Let me illustrate: 
![DrawFarmland](https://github.com/user-attachments/assets/a07d19c1-135e-40b3-b1cd-02cf7853bd95)
I'm drawing a farmland here, `f`ollowing the existing forest boundary. When I encounter a tripoint with a meadow, as in the screenshot, I want to continue `f`ollowing the meadow. Therefore, I _want_ iD to stop there and prompt me where to continue. Implementing this change would mean that it would just continue following the forest (something I rarely if ever do), so it will make my workflow very error-prone.

**Comment by 1ec5** [comment]
The current behavior makes the follow operation useless in some common situations. For example, I happen to know that there’s a fence surrounding this quarry, which is thoroughly connected to surrounding landuse areas. I can’t use the follow operation at all, because every single node is connected to more than one area. Yet it doesn’t even matter which area it follows:

![Drawing a line along the boundary between a quarry landuse area and a woods landuse area.](https://github.com/user-attachments/assets/5edaa568-218a-4cb7-9834-4ce937fb039e)

I take it you’re jamming the <kbd>F</kbd> key until it pauses on something. I do that a lot when editing OpenHistoricalMap, in order to duplicate a way and ensure that it’s connected to the original way. It’s a critical workflow for that project, but it’s also hampered by this issue. What if we initially pause but continue along the focused way after the second keystroke? Or what if we detect that the <kbd>F</kbd> key is on repeat and pause for a second or two, giving you a chance to ease up on the <kbd>F</kbd> key?

#10155 also had a good idea about considering the angle of each candidate way.

**Comment by DujaOSM** [comment]
But in the fence use case, it's just the software being lame: yeah, the current point connects multiple areas, but they _share both the previous and next node_, so there's no ambiguity. The algorithm could be amended to work in that case as well, rather than changing the whole workflow.

(And, btw, related #9734 could be tackled along the way, seems simple enough).

**Comment by matkoniecz** [comment]
@DujaOSM 

> Implementing this change would mean that it would just continue following the forest (something I rarely if ever do), so it will make my workflow very error-prone.

this seems entirely solvable by looking at screen and stopping to hold `f` or stopping to repeatedly press `f` when you reach location where you want to branch off? Or, in the worst case - undoing few operations?

**Comment by DujaOSM** [comment]
@matkoniecz 
| Invoking the follow operation would focus the way, and subsequent follow operations would choose that way from among the candidate ways.

this seems entirely solvable (as it is currently working) by looking at screen, and when the operation gets stuck, click on the next node you want to follow. [See what I did here? :) ] 

I map **a lot** of landcover, and I encounter this choice about two dozen times a day. The proposed change would badly hamper my workflow, forcing me to pay a lot more attention and backspace every now and then. 

Seriously, as a middle ground, I think that the use case from Minh's [last post](https://github.com/openstreetmap/iD/issues/8750#issuecomment-2414935743), where the next point is the part of both ways, can be uncontroversially solved, and let that be the end of it.
